tests/chainer_tests/functions_tests/math_tests/test_tensordot.py

```diff
@@ -0,0 +1,141 @@
+        {'a_shape': (3, 5, 2, 4), 'b_shape': (6, 3, 2), 'axes': ([0, 2], [1, 2]), 'gc_shape': (5, 4, 6)},  # NOQA
+        {'a_shape': (5, 3, 2, 4), 'b_shape': (6, 2, 3), 'axes': ([1, 2], [2, 1]), 'gc_shape': (5, 4, 6)},  # NOQA
+
+        {'a_shape': (5, 4, 3, 2), 'b_shape': (4, 3, 2, 6), 'axes': 3, 'gc_shape': (5, 6)},  # NOQA
+        {'a_shape': (5, 4, 3, 2), 'b_shape': (4, 3, 2, 6), 'axes': ([1, 2, 3], [0, 1, 2]), 'gc_shape': (5, 6)},  # NOQA
+        {'a_shape': (5, 4, 2, 3), 'b_shape': (4, 3, 6, 2), 'axes': ([1, 3, 2], [0, 1, 3]), 'gc_shape': (5, 6)},  # NOQA
+        {'a_shape': (5, 2, 4, 3), 'b_shape': (4, 6, 3, 2), 'axes': ([2, 3, 1], [0, 2, 3]), 'gc_shape': (5, 6)},  # NOQA
+        {'a_shape': (2, 5, 4, 3), 'b_shape': (4, 6, 2, 3), 'axes': ([2, 3, 0], [0, 3, 2]), 'gc_shape': (5, 6)},  # NOQA
+        {'a_shape': (2, 5, 3, 4), 'b_shape': (6, 4, 2, 3), 'axes': ([3, 2, 0], [1, 3, 2]), 'gc_shape': (5, 6)},  # NOQA
+        {'a_shape': (2, 3, 5, 4), 'b_shape': (6, 2, 4, 3), 'axes': ([3, 1, 0], [2, 3, 1]), 'gc_shape': (5, 6)},  # NOQA
+        {'a_shape': (3, 2, 5, 4), 'b_shape': (6, 2, 3, 4), 'axes': ([3, 0, 1], [3, 2, 1]), 'gc_shape': (5, 6)},  # NOQA
+        {'a_shape': (3, 2, 4, 5), 'b_shape': (2, 6, 3, 4), 'axes': ([2, 0, 1], [3, 2, 0]), 'gc_shape': (5, 6)},  # NOQA
+        {'a_shape': (3, 4, 2, 5), 'b_shape': (2, 3, 6, 4), 'axes': ([1, 0, 2], [3, 1, 0]), 'gc_shape': (5, 6)},  # NOQA
+        {'a_shape': (4, 3, 2, 5), 'b_shape': (2, 3, 4, 6), 'axes': ([0, 1, 2], [2, 1, 0]), 'gc_shape': (5, 6)},  # NOQA
+        {'a_shape': (4, 3, 5, 2), 'b_shape': (3, 2, 4, 6), 'axes': ([0, 1, 3], [2, 0, 1]), 'gc_shape': (5, 6)},  # NOQA
+        {'a_shape': (4, 5, 3, 2), 'b_shape': (3, 4, 2, 6), 'axes': ([0, 2, 3], [1, 0, 2]), 'gc_shape': (5, 6)},  # NOQA
+    ],
+    [
+        {'a_dtype': numpy.float16},
+        {'a_dtype': numpy.float32},
+        {'a_dtype': numpy.float64},
+    ],
+    [
+        {'b_dtype': numpy.float16},
+        {'b_dtype': numpy.float32},
+        {'b_dtype': numpy.float64},
+    ]
+))
+class TestTensorDot(unittest.TestCase):
+
+    def setUp(self):
+        self.a = numpy.random.uniform(.5, 1, self.a_shape)
+        self.a = self.a.astype(self.a_dtype)
+        self.b = numpy.random.uniform(.5, 1, self.b_shape)
+        self.b = self.b.astype(self.b_dtype)
+        ret_dtype = numpy.result_type(self.a_dtype, self.b_dtype)
+        self.gc = numpy.random.uniform(-1, 1, self.gc_shape).astype(ret_dtype)
+        self.gga = numpy.random.uniform(
+            .5, 1, self.a_shape).astype(self.a_dtype)
+        self.ggb = numpy.random.uniform(
+            .5, 1, self.b_shape).astype(self.b_dtype)
+
+        self.op = lambda a, b: F.tensordot(a, b, axes=self.axes)
+        self.forward_answer = self._get_forward_answer(self.a, self.b,
+                                                       self.axes)
+
+    def _get_forward_answer(self, a, b, axes):
```

Comment:
I see that this method is aligned to other tests but I don't think it's needed for tensor dot (since the variable name is descriptive and there are no branches).